### PR TITLE
Fix scout killing (AO-13011, AO-13015)

### PR DESCRIPTION
**Pull request by lootek**
JIRA: AO-13011
JIRA: AO-13015

### CODE REVIEW

File: lib/scout/scout.rb [hunk 1]
```diff
@@ -52,6 +39,25 @@ def fix_permissions
       system "chmod -v g+rw /var/log/scout/scoutd.log"
       system "chmod -Rv g+w /var/lib/scoutd"
     end
+
+    private
+
+    def wait_for_scoutd_and_children(scoutd_pid, signal = 15)
```
Comment (gortyy) [inline lib/scout/scout.rb]: `signal` could be string, would be more readable, in this case `"TERM"`

File: lib/scout/scout.rb [hunk 2]
```diff
@@ -17,26 +17,13 @@ def deactivate
...
-          system "kill -9 $(pgrep #{SCOUTD}) 2>/dev/null" if system "pgrep #{SCOUTD}"
-        end
+      # term
+      wait_for_scoutd_and_children(scoutd_pid, 15)
 
-        i += 1
-        sleep 1
-      end
+      # kill
+      wait_for_scoutd_and_children(scoutd_pid, 9)
```
Comment (gortyy) [inline lib/scout/scout.rb]: `"KILL"` instead of `9`

File: lib/scout/scout.rb (hunk 1 above)
Comment (lootek) [reply lib/scout/scout.rb]: 👍

File: lib/scout/scout.rb (hunk 2 above)
Comment (lootek) [reply lib/scout/scout.rb]: also 👍

**Pull request merged**